examples/seq2seq/seq2seq_trainer.py

```diff
@@ -60,6 +59,11 @@ def __init__(self, config=None, data_args=None, *args, **kwargs):
                 self.config.pad_token_id is not None
             ), "Make sure that `config.pad_token_id` is correcly defined when ignoring `pad_token` for loss calculation or doing label smoothing."
 
+        if self.config.pad_token_id is None and self.config.eos_token_id is not None:
+            logger.warn(
+                f"The `config.pad_token_id` is `None`. Using `config.eos_token_id` = {self.config.eos_token_id} for padding.."
+            )
```

Comment:
what if `eos_token_id is None`? Should we raise?

Reply:
should we raise early in that case?

Reply:
What I meant is that there are models, like `openai-gpt` or `ctrl` that do not have a `eos_token_id` nor do they have a `pad_token_id` => the way it is implemented now these models could still make use of seq2seqTrainer because they would never require padding (because they never finish early). So I'd just leave it as it is - or if you think that models that don't have an EOS token should not use Seq2SeqTrainer we could raise as well - up to you!

Reply:
Didn't understand that they always go to `max_length` your implem makes total sense. Thanks for clarifying.